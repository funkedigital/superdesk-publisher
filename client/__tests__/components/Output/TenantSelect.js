import React from "react";
import TenantSelect from "../../../components/Output/TenantSelect";
import { render, fireEvent, wait } from "@testing-library/react";
import Store from "../../../components/Output/Store";


describe("Output/TenantSelect", () => {

  it("renders correctly", () => {
    const { container } = render(
      <Store.Provider
        value={{
          tenants: [
            { name: 'tenant1', code: 'tenant1' },
            { name: 'tenant2', code: 'tenant2' }
          ],
          filters: {},
          actions: { setFilters: jest.fn() }
        }}
      >
        <TenantSelect />
      </Store.Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("sets tenant", async () => {
    const setFilters = jest.fn();

    const { getByText } = render(
      <Store.Provider
        value={{
          tenants: [
            { name: 'tenant1', code: 'tenant1' },
            { name: 'tenant2', code: 'tenant2' }
          ],
          filters: {},
          actions: { setFilters: setFilters }
        }}
      >
        <TenantSelect />
      </Store.Provider>
    );

    const button = getByText('tenant2');
    fireEvent.click(button);

    expect(setFilters).toHaveBeenCalled();
    expect(setFilters).toHaveBeenCalledWith({ tenant: { name: 'tenant2', code: 'tenant2' } });
  });
});
