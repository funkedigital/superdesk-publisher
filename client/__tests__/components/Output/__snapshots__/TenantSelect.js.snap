// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output/TenantSelect renders correctly 1`] = `
<div
  class="subnav__spacer subnav__spacer--no-margin"
/>
`;
